src/puj/push/unit.clj

```diff
@@ -71,14 +71,29 @@
 
             :else
             (-> state
-                (state/pop-from-stacks input-stacks)
-                (state/push-to-stacks
+                (pushstate/pop-from-stacks input-stacks)
+                (pushstate/push-to-stacks
                  ; push-to-stacks expects a collection. If results is not a collection, make it one.
                  (make-collection results)
                  ; same with output-stacks
                  (make-collection output-stacks)))))))))
 
 
+(defn simple-instruction
+  "Creates a SimpleInstruction which uses a standardized way of manipulating Push states. In other words, it
+  handles popping its own function arguments and pushing the function return values.
+
+  The first step of evaluating a SimpleInstruction is to pop the arguments from the stacks corresponding
+  to the instruction's `input-stacks` list. If multiple occurrences of the same type are in `input_stacks`,
+  items are taken from progressively deeper in that stack. If the stacks of the  PushState do not contain
+  a sufficient number of items, the instruction does not modify the PushState.
+
+  The popped arguments are then passed to the instruction's function to produce the return value(s).
+  The return value(s) are then routed to the corresponding stacks specified in the instruction's `output-stacks`."
```

Comment:
It might be nice to add a note here that `output-stacks` can be either a collection of stacks or a single stack.